compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmTerminatorGen.java

```diff
@@ -733,7 +733,8 @@ private void genCall(BIRTerminator.Call callIns, PackageID packageID, int localV
         }
 
         BIRNode.BIRVariableDcl selfArg = callIns.args.get(0).variableDcl;
-        if (JvmCodeGenUtil.getReferredType(selfArg.type).tag == TypeTags.OBJECT) {
+        BType selfArgRefType = JvmCodeGenUtil.getReferredType(selfArg.type);
+        if (selfArgRefType.tag == TypeTags.OBJECT  || selfArgRefType.tag == TypeTags.UNION) {
```

Comment:
Do we know for what kind of call instructions we get to the else block here? Just wondering if we can get there for non-object unions, in which case this change will evaluate to true here and cause wrong code generation.

Reply:
https://github.com/ballerina-platform/ballerina-lang/blob/master/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmTerminatorGen.java#L746 This line is not reached by any of the unit-tests, Do we need this if else check here? @warunalakshitha @manuranga

Reply:
It looks like the only possible types reach here are Ballerina objects and union of objects. Compiler won't allow to write a program where the `selfArgRefType` is a non-object union. So IMO, there are no possibilities to a wrong code generation. 
Furthermore, It seems the else part is no longer in use. We can create a separate task to remove unused lines.

Reply:
@prakanth97 please create a task for this just to track this

Reply:
Created https://github.com/ballerina-platform/ballerina-lang/issues/38599